src/sage/rings/function_field/khuri_makdisi.pyx

```diff
@@ -838,13 +838,36 @@ cdef class KhuriMakdisi_small(KhuriMakdisi_base):
             sage: q2 = G.point(pl2 - b)
             sage: G.point(af.divisor()) == q1.addflip(p2)
             True
+
+        Check that :issue:`40237` is fixed::
+
+            sage: K = GF(2)
+            sage: F.<x> = FunctionField(K)
+            sage: t = polygen(F)
+            sage: E.<y> = F.extension(t^3 + (x^2 + x + 1)*t^2 + (x^3 + x + 1)*t + x^5 + x^4)
+            sage: O = E.maximal_order()
+            sage: Oinf = E.maximal_order_infinite()
+            sage: D1 = (-5 * O.ideal(x, y).divisor() + O.ideal(x + 1, y^2 + y + 1).divisor()
+            ....:   + O.ideal(x^3 + x^2 + 1, y + x + 1).divisor())
+            sage: D2 = (Oinf.ideal(1/x, y/x^2 + 1).divisor() - 5 * O.ideal(x, y).divisor()
+            ....:   + O.ideal(x^4 + x^3 + 1, y + x).divisor())
+            sage: assert D1.degree() == 0
+            sage: assert D2.degree() == 0
+            sage: J = E.jacobian('km_small')
```

Comment:
```suggestion
            sage: J = E.jacobian('km_small', base_div=5 * O.ideal(x, y).divisor())
```

Since this test depends on the base divisor, let's specify it to make sure this case is still tested if the way the default base divisor is chosen ever changes in the future.

Reply:
Right. Thanks. Done.